### DISCUSSION TITLE
Evaluate keycloak Authentication

### DISCUSSION THREAD

**Discussion by rsoika** [imixs/imixs-workflow#730] [Ideas]
To run the docker container

`docker run -p 8080:8080 -p 9990:9990 -e KEYCLOAK_USER=admin -e KEYCLOAK_PASSWORD=xxxx jboss/keycloak`

Start web application from
http://localhost:8080/auth/

For saml configuration see:
https://keycloak.gitbooks.io/securing-client-applications-guide/content/topics/saml/java/jboss-adapter/securing_wars.html

**Comment by shaun-forgie** [comment]
Is Keycloak integration being looked at? This would be a fantastic value add to the project and expose it to a much larger community.

**Comment by rsoika** [comment]
@shaun-forgie yes, it is possible to use Keycloak with imixs-worklfow.
We have written a short tutorial about the integration: 
https://blog.imixs.org/2016/12/10/integrate-imixs-workflow-single-sign/

The tutorial is a bit older, so details may have changed slightly. But technically, it works great.

**Comment by shaun-forgie** [comment]
@rsoika Is there support for pulling in role assignments defined in Keycloak? For example as discussed here https://www.imixs.org/doc/quickstart/roles_responsibilities.html ?

**Comment by rsoika** [comment]
Hi, there is a different between the access roles and the organizational roles within a specific business process. The link is about the later scenario. 
In general it is sufficient if you authenticate users against one of the [Imixs Access roles](https://www.imixs.org/doc/deployment/security.html) 

- org.imixs.ACCESSLEVEL.READACCESS
- org.imixs.ACCESSLEVEL.AUTHORACCESS
- org.imixs.ACCESSLEVEL.EDITORACCESS
- org.imixs.ACCESSLEVEL.MANAGERACCESS

This can be done by Keycloak. Within a specific workflow process or your application you can resolve later organizational roles in different ways (e.g. from a ldap) or within your application as done by Imixs Office-Workflow 

![image](https://user-images.githubusercontent.com/1339124/196900697-843ee347-e735-4092-b234-0fafa3952a93.png)

Imixs-Workflow engine takes care about those kind of roles and can protect your business data individual.
